plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -106,6 +107,33 @@ private static void checkIsDoubled(ZoneId zone, LocalDateTime dateTime)
         verify(zone.getRules().getValidOffsets(dateTime).size() == 2, "Expected %s to be doubled in %s", dateTime, zone);
     }
 
+    @Test
+    public void testTrinoBoolean()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("boolean", "1", SMALLINT, "SMALLINT '1'")
+                .addRoundTrip("boolean", "0", SMALLINT, "SMALLINT '0'")
+                .addRoundTrip("boolean", "NULL", SMALLINT, "CAST(NULL AS SMALLINT)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_boolean"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_boolean"));
+    }
+
+    @Test
+    public void testBool()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("bool", "true", BOOLEAN, "BOOLEAN '1'")
+                .addRoundTrip("bool", "false", BOOLEAN, "BOOLEAN '0'")
+                .addRoundTrip("Nullable(bool)", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
+                .execute(getQueryRunner(), clickhouseCreateAndInsert("tpch.test_boolean"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("bool", "true", BOOLEAN, "BOOLEAN '1'")
+                .addRoundTrip("bool", "false", BOOLEAN, "BOOLEAN '0'")
+                .addRoundTrip("Nullable(bool)", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
+                .execute(getQueryRunner(), clickhouseCreateAndTrinoInsert("tpch.test_boolean"));
```

Comment:
We prefer `true` to `BOOLEAN '1'` and there is no need to separate SqlDataTypeTest:
```java
        SqlDataTypeTest.create()
                .addRoundTrip("bool", "true", BOOLEAN, "true")
                .addRoundTrip("bool", "false", BOOLEAN, "false")
                .addRoundTrip("Nullable(bool)", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
                .execute(getQueryRunner(), clickhouseCreateAndInsert("tpch.test_boolean"))
                .execute(getQueryRunner(), clickhouseCreateAndTrinoInsert("tpch.test_boolean"));
```